### PR TITLE
chore: koala identify call

### PR DESCRIPTION
**Pull request by thisisnithin**
## Motivation and Context

Identifies users on Koala.

Completes ENG-4941

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: studio/src/components/app-provider.tsx [hunk 1]
```diff
@@ -164,6 +170,17 @@ export const AppProvider = ({ children }: { children: ReactNode }) => {
         invitations: data.invitations,
       });
 
+      window.ko?.identify(data.email, {
```
Comment (StarpTech) [inline studio/src/components/app-provider.tsx:173]: Can we make it more obvious that this only works when the script was provided? Maybe add a condition around with a few comments?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: studio/src/components/app-provider.tsx (hunk 1 above)
Comment (thisisnithin) [reply studio/src/components/app-provider.tsx]: Done. Added reset too

File: studio/src/components/app-provider.tsx [hunk 2]
```diff
@@ -164,6 +174,20 @@ export const AppProvider = ({ children }: { children: ReactNode }) => {
...
+      // If custom head scripts are available, enable identify call for koala script
+      if (process.env.CUSTOM_HEAD_SCRIPTS) {
+        window.ko?.identify(data.email, {
+          id: data.id,
+          $account: {
+            organizationId: organization.id,
+            organizationName: organization.name,
+            organizationSlug: organization.slug,
+            plan: organization.plan,
+            roles: organization.roles,
```
Comment (StarpTech) [inline studio/src/components/app-provider.tsx:186]: We don't need to send the roles.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: studio/src/components/app-provider.tsx [hunk 3]
```diff
@@ -93,6 +99,10 @@ const fetchSession = async () => {
     }
     return null;
   } catch (e) {
+    // Reset koala if custom head scripts are found and user is not authenticated
+    if (process.env.CUSTOM_HEAD_SCRIPTS) {
+      window.ko?.reset();
```
Comment (StarpTech) [inline studio/src/components/app-provider.tsx:104]: We also need to call `reset` when the user clicks on logout.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: studio/src/components/app-provider.tsx (hunk 2 above)
Comment (thisisnithin) [reply studio/src/components/app-provider.tsx]: Removed.

File: studio/src/components/app-provider.tsx (hunk 3 above)
Comment (thisisnithin) [reply studio/src/components/app-provider.tsx:104]: Done.

**Review by StarpTech — APPROVED** [review]
Lgtm

**Pull request merged**